### DISCUSSION TITLE
rpm changed some bytes of statically linked library

### DISCUSSION THREAD

**Discussion by shunboli** [rpm-software-management/rpm#2313] [Q&A] [answered]
Firstly, i packed all the files in the main directory 1, and then extracted them to a new directory 2; 
Secondly, i added one file in the main directory 1 and pack them again, then extracted them to a new directory 3;
the same statically linked library file in directory 2 and directory 3 is not the same when i use vimdiff, some bytes was changed, but when i use readelf -a, the results are the same, why?  and what was changed?
%debug_package %{nil} was set in .spec file.  but the statically linked library file does not have debuginfo（i used readelf -S to read it)

**Comment by pmatilai** [comment, marked as answer]
Build root policy scripts (that brp-* stuff) can and do modify build root contents, sometimes inadvertely.
I'd hazard a guess this is brp-strip-static-archive in action, *assuming* this is merely upstream brp-scripts. Practically all distros supply their own, override/replace upstream ones etc so impossible to say with this level of detail.

**Reply by shunboli** [reply]
I did another test, pack the same files in directory1 twice, then extract them to different directory, There is still a difference between the static library file，only a few bytes was changed, i use %!xxd to convert binary to hexadecimal，here are the header of the two files, Indicate distinction by bolding them, do you know what was changed, time？or something else? (I add some space in the first line of the two file or  they will be  hidden in this reply )
first file:
! < arch > 
 /              16711557**68**  0     0     0       29038   
 0000000: 213c 6172 6368 3e0a 2f20 2020 2020 2020  !<arch
 0000010: 2020 2020 2020 2020 3136 3731 3135 3537        
 0000020: **3638** 2020 3020 2020 2020 3020 2020 2020  **68**  0
second file:
! < arch >
/               16711557**20**  0     0     0       29038
0000000: 213c 6172 6368 3e0a 2f20 2020 2020 2020  !<arch
0000010: 2020 2020 2020 2020 3136 3731 3135 3537
0000020: **3230** 2020 3020 2020 2020 3020 2020 2020  **20**  0

**Comment by pmatilai** [comment]
Yes, they're different. Like said, they're almost certainly brp scripts in action, no deep mystery here.
